### PR TITLE
Add Linting

### PR DESCRIPTION
**Pull request by wiese**
* eslint
* stylelint
* travis integration
* run tests in CI

Potential merge conflicts: #27, #29

### CODE REVIEW

File: desktop/css/wlightbox.css [hunk 1]
```diff
@@ -62,7 +62,7 @@
 
 #wlightbox:before {
 	border: solid transparent;
-	content: " ";
+	content: '';
```
Comment (KaiNissen) [inline desktop/css/wlightbox.css]: the whitespace vanished

File: mobile/css/styles.pcss [hunk 2]
```diff
@@ -676,7 +676,8 @@ li.subhead {
 @media ( min-width: 800px ) {
 	#frbanner-window .a-sms,
 	#frbanner-window button {
-		height: 48px; /*font-size: 1.1em;*/;
+		height: 48px;
+		/* font-size: 1.1em; */
```
Comment (KaiNissen) [inline mobile/css/styles.pcss]: Could as well be removed. This is not a comment, but a "cautious" change.

**Comment by wiese** [comment]
As discussed w/ @KaiNissen, switched colors to long hex form against the rules defined by https://github.com/wikimedia/stylelint-config-wikimedia

File: mobile/banner_ctrl.js [hunk 3]
```diff
@@ -118,20 +118,20 @@ $( '#btn-bez' ).click( function () {
...
 function debounce( func, wait, immediate ) {
 	var timeout;
-	return function() {
+	return function () {
 		var context = this, args = arguments;
-		var later = function() {
+		var later = function () {
 			timeout = null;
-			if ( !immediate ) func.apply( context, args );
+			if ( !immediate ) { func.apply( context, args ); }
```
Comment (KaiNissen) [inline mobile/banner_ctrl.js]: autofix failed

File: mobile/banner_ctrl.js [hunk 4]
```diff
@@ -118,20 +118,20 @@ $( '#btn-bez' ).click( function () {
...
 		};
 		var callNow = immediate && !timeout;
 		clearTimeout( timeout );
 		timeout = setTimeout( later, wait );
-		if ( callNow ) func.apply( context, args );
+		if ( callNow ) { func.apply( context, args ); }
```
Comment (KaiNissen) [inline mobile/banner_ctrl.js]: autofix failed

File: mobile/banner_var.js [hunk 5]
```diff
@@ -119,20 +119,20 @@ $( '#btn-bez' ).click( function () {
...
 function debounce( func, wait, immediate ) {
 	var timeout;
-	return function() {
+	return function () {
 		var context = this, args = arguments;
-		var later = function() {
+		var later = function () {
 			timeout = null;
-			if ( !immediate ) func.apply( context, args );
+			if ( !immediate ) { func.apply( context, args ); }
```
Comment (KaiNissen) [inline mobile/banner_var.js]: autofix failed

File: mobile/banner_var.js [hunk 6]
```diff
@@ -119,20 +119,20 @@ $( '#btn-bez' ).click( function () {
...
 		};
 		var callNow = immediate && !timeout;
 		clearTimeout( timeout );
 		timeout = setTimeout( later, wait );
-		if ( callNow ) func.apply( context, args );
+		if ( callNow ) { func.apply( context, args ); }
```
Comment (KaiNissen) [inline mobile/banner_var.js]: autofix

File: shared/custom_day_name.js [hunk 7]
```diff
@@ -1,19 +1,19 @@
...
 		de: 'Weihnachtsfeiertag',
 		en: 'Christmas Day'
 	},
-	'25.12': {
+	25.12: {
 		de: '1. Weihnachtsfeiertag',
 		en: 'Christmas Day'
 	},
-	'26.12': {
+	26.12: {
```
Comment (KaiNissen) [inline shared/custom_day_name.js]: The autofix did not remove the quotes of the first key. Is it actually good practice to use an unquoted value containing a period?

File: mobile/banner_ctrl.js [hunk 8]
```diff
@@ -198,6 +198,10 @@ $( document ).ready( function () {
 	}
 } );
 
+function addPointsToNum() {
+	// fixme Called in animateProgressBar() but not implemented!
```
Comment (KaiNissen) [inline mobile/banner_ctrl.js]: I guess, this will be part of the progress bar refactoring.

**Review by KaiNissen — COMMENTED** [review]
Generally approving. Follow-up on some issues in #35.

**Pull request merged**